orchestration/taskdefinition.go

```diff
@@ -110,7 +121,59 @@ func (o *Orchestrator) processTaskDefinitionUpdate(ctx context.Context, input *S
 		input.TaskDefinition.NetworkMode = DefaultNetworkMode
 	}
 
+	logConfiguration, err := o.processLogConfiguration(ctx, input.ClusterName, aws.StringValue(input.TaskDefinition.Family), input.TaskDefinition.Tags)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, cd := range input.TaskDefinition.ContainerDefinitions {
+		cd.SetLogConfiguration(logConfiguration)
+	}
+
 	log.Infof("creating task definition %+v", input.TaskDefinition)
 
 	return o.ECS.CreateTaskDefinition(ctx, input.TaskDefinition)
 }
+
+func (o *Orchestrator) processLogConfiguration(ctx context.Context, logGroup, streamPrefix string, tags []*ecs.Tag) (*ecs.LogConfiguration, error) {
+	if logGroup == "" {
+		return nil, errors.New("cloudwatch logs group name cannot be empty")
+	}
+
+	var tagsMap = make(map[string]*string)
+	for _, tag := range tags {
+		tagsMap[aws.StringValue(tag.Key)] = tag.Value
+	}
+
+	err := o.CloudWatchLogs.CreateLogGroup(ctx, &cloudwatchlogs.CreateLogGroupInput{
+		LogGroupName: aws.String(logGroup),
+		Tags:         tagsMap,
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case cloudwatchlogs.ErrCodeResourceAlreadyExistsException:
+				log.Warnf("cloudwatch log group already exists, continuing: (%s)", err)
```

Comment:
We might want to do something if it's an existing log group, e.g. update the tags and retention (I'm thinking when this gets executed during taskdef update)

Reply:
👍